src/component/Text.tsx

```diff
@@ -183,109 +183,113 @@ const getWordsByLines = (props: Props, needCalculate: boolean) => {
   return getWordsWithoutCalculate(props.children);
 };
 
-export class Text extends Component<Props, State> {
-  static defaultProps = {
-    x: 0,
-    y: 0,
-    lineHeight: '1em',
-    capHeight: '0.71em', // Magic number from d3
-    scaleToFit: false,
-    textAnchor: 'start',
-    verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
-    fill: '#808080',
-  };
+const textDefaultProps = {
+  x: 0,
+  y: 0,
+  lineHeight: '1em',
+  capHeight: '0.71em', // Magic number from d3
+  scaleToFit: false,
+  textAnchor: 'start',
+  verticalAnchor: 'end', // Maintain compat with existing charts / default SVG behavior
+  fill: '#808080',
+};
 
-  state: State = {};
+export const Text = (props: Props) => {
+  const [state, setState] = useState<State>({
+    prevWidth: props.width,
+    prevScaleToFit: props.scaleToFit,
+    prevChildren: props.children,
+    prevStyle: props.style,
+    wordsByLines: getWordsByLines(props, true),
+  });
 
-  static getDerivedStateFromProps(nextProps: Props, prevState: State): State {
+  useEffect(() => {
     if (
-      nextProps.width !== prevState.prevWidth ||
-      nextProps.scaleToFit !== prevState.prevScaleToFit ||
-      nextProps.children !== prevState.prevChildren ||
-      nextProps.style !== prevState.prevStyle ||
-      nextProps.breakAll !== prevState.prevBreakAll
+      props.width !== state.prevWidth ||
+      props.scaleToFit !== state.prevScaleToFit ||
+      props.children !== state.prevChildren ||
+      props.style !== state.prevStyle ||
+      props.breakAll !== state.prevBreakAll
     ) {
       const needCalculate =
-        nextProps.children !== prevState.prevChildren ||
-        nextProps.style !== prevState.prevStyle ||
-        nextProps.breakAll !== prevState.prevBreakAll;
-
-      return {
-        prevWidth: nextProps.width,
-        prevScaleToFit: nextProps.scaleToFit,
-        prevChildren: nextProps.children,
-        prevStyle: nextProps.style,
-        wordsByLines: getWordsByLines(nextProps, needCalculate),
-      };
+        props.children !== state.prevChildren ||
+        props.style !== state.prevStyle ||
+        props.breakAll !== state.prevBreakAll;
+
+      setState({
+        prevWidth: props.width,
+        prevScaleToFit: props.scaleToFit,
+        prevChildren: props.children,
+        prevStyle: props.style,
```

Comment:
I am not sure I understand why these are in the state at all. 🤔 
Would it not be sufficient to only have them as props?

Reply:
I am raising this as a question, not as a thing to fix. Your PR already adds value and seems good to go.

Reply:
I dig a bit and all those was for computing whether we should do a recalculation or not, I simplified the code and removed those states, a useMemo would be sufficient IMO.